### PR TITLE
New language: Esperanto

### PR DESCRIPTION
**Pull request by babynanxi**
New language: Esperanto
There may be a problem with some syntax

### CODE REVIEW

**Comment by LNSSPsd** [comment]
Improvements might be needed as it couldn't pass ChatGPT's translation quality test.
Here's the detail:
```
The quality of the translation appears to be average. Some of the translations are accurate and convey the intended meaning, while others are not as clear or contain grammar errors. The use of Esperanto appears to be inconsistent and may be confusing for some users. It is recommended to review and improve the translations for clarity and consistency.
```
`- Can you give some examples on inappropriate translations?`
```
Sure! Here are a few examples of translations that could use improvement:

ACME_FailedToGetCommand: "Malsukcesis ricevi acme komandon." This translation is grammatically correct but could be more clear. A better translation might be: "La akme komando malsukcesis esti ricevita."
Auth_InvalidFBVersion: "Nevalida FastBuilder-versio, bonvolu ĝisdatigi." The use of Esperanto here is inconsistent with the rest of the translations and may be confusing for some users. A better translation might be: "Nevalida versio de FastBuilder, bonvolu ĝisdatigi."
Auth_ServerNotFound: "Servilo ne trovita, bonvolu kontroli la publikan staton de via servilo" This translation contains a grammar error and could be more clear. A better translation might be: "La servilo ne estas trovita, bonvolu kontroli la publikan staton de via servilo."
These are just a few examples, and there may be other translations in the text that could use improvement as well.
```

However, what a language model says couldn't always be seen as truth, you can insist on that yours is better and this is only for a reference.

**Comment by Torrekie** [comment]
we need to find somebody who learned this language hmm

**Comment by babynanxi** [comment]
I'm getting dizzy. Although Esperanto is relatively simple, I can't

**Comment by SuperScript-PRC** [comment]
Oh，这是南希写的啊..？

**Comment by babynanxi** [comment]
I reviewed it three times and personally I am sure that there is no problem, please do the quality test again

File: fastbuilder/i18n/i18n.go [hunk 1]
```diff
@@ -16,6 +16,7 @@ const (
 	LanguageTraditionalChinese = "zh_HK"
 	LanguageTaiwanChinese      = "zh_TW"
 	LanguageJapanese           = "ja_JP"
+	LanguageEsperanto          = "epo"
```
Comment (Torrekie) [inline fastbuilder/i18n/i18n.go:19]:
```suggestion
	LanguageEsperanto          = "eo"
```

File: fastbuilder/i18n/i18n.go [hunk 2]
```diff
@@ -178,6 +179,7 @@ var LangDict map[string]map[uint16]string = map[string]map[uint16]string{
 	LanguageTraditionalChinese: I18nDict_zh_HK,
 	LanguageTaiwanChinese:      I18nDict_zh_TW,
 	LanguageJapanese:           I18nDict_ja_JP,
+	LanguageEsperanto:          I18nDict_epo,
```
Comment (Torrekie) [inline fastbuilder/i18n/i18n.go:182]:
```suggestion
	LanguageEsperanto:          I18nDict_eo,
```

**Review by Torrekie — CHANGES_REQUESTED** [review]
According to the *nix locale system, Esperanto should be `eo` but not `epo`

Remember rename epo.go to eo.go

**Comment by babynanxi** [comment]
OK

**Comment by LNSSPsd** [comment]
Please keep constant strings (arguments) constant.
For an instance, the building option `discrete` should be `discrete` among all the languages,
here's what in your translation:
`DelayThreshold_OnlyDiscrete: "Prokrasta sojlo disponeblas nur kun prokrasta reĝimo: diskreta",`
Users have no way to access that option by giving `diskreta`.

**Comment by babynanxi** [comment]
sorry wrong operation

**Comment by LNSSPsd** [comment]
Here's what in your updated version:
```
DelayThreshold_OnlyDiscrete: "Prokrastsojlo disponeblas nur kun prokrasta reĝimo: disperso",
```
However, possible options for delay threshold are only `continuous`, `discrete` or `none`, `disperso` isn't going to be one, so it's better to keep it constant.

**Comment by babynanxi** [comment]
Is "discrete" a value?

**Comment by LNSSPsd** [comment]
> Is "discrete" a value?

It's an enum value.

**Comment by babynanxi** [comment]
Restore the enum value

**Comment by LNSSPsd** [comment]
And DelaySetUnavailableUnderNoneMode, etc.

**Comment by LNSSPsd** [comment]
Some more problems figured out by ChatGPT:
```
Based on my assessment of the translation, there are a few errors that need to be addressed. Here's a breakdown of the errors:

    The word "Enter" has been translated as "Enigu", which is not a correct translation. A better translation for "Enter" in this context would be "Enmetu".
    The word "Password" has been translated as "Pasvorton", which is correct, although it could also be translated as "Pasvorto" without changing the meaning significantly.
    The phrase "Press [Enter]" has been translated as "Premu [Enigu]", which is not a correct translation. A better translation for "Press [Enter]" in this context would be "Premu [Enmetu]".
    The phrase "if not set" has been translated as "se ne agordita", which is not a correct translation. A better translation for "if not set" in this context would be "se ne difinita".
    The phrase "input won't be echoed" has been translated as "enigo ne estos ripetita", which is not a correct translation. A better translation for "input won't be echoed" in this context would be "eniga teksto ne estos montrita".

Overall, I would give the translation a score of 60 out of 100, as there are several errors in the translation that could cause confusion for the reader. A more accurate translation could be: "Enmetu Pasvorton (Premu [Enmetu] se ne difinita, eniga teksto ne estos montrita):".
```

**Comment by shwodbowhsjdk** [comment]
为什么你们用英文交流啊，不都要翻译回来吗What are you pretending to be, man

**Comment by Happy2018new** [comment]
> 为什么你们用英文交流啊，不都要翻译回来吗What are you pretending to be, man

习惯吧

**Comment by SuperScript-PRC** [comment]
>  为什么你们用英文交流啊，不都要翻译回来吗What are you pretending to be, man
因为here's github

**Comment by Inotart** [comment]
草

**Comment by babynanxi** [comment]
> 

？

**Comment by Inotart** [comment]
你好,我目前不在，如果你只是看到论坛邮件就不用回复了

**Pull request closed without merging**